### PR TITLE
Adjust RolloverAction serialization of maxPrimaryShardSize during an upgrade

### PR DESCRIPTION
**Pull request by martijnvg**
If node doesn't support maxPrimaryShardSize then serialize maxPrimaryShardSize as maxSize.
This should fix a problematic situation if an older node doesn't support maxPrimaryShardSize
and this is the only condition specified then the older node ends up with a instance without
any conditions. This could lead to upgrade failures, new nodes not able to start because
local cluster state can't be read.

Relates to #69918 and #69995

(marking as non issue, since #69995 and #68917 have not been released yet)

### CODE REVIEW

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/RolloverAction.java [hunk 1]
```diff
@@ -96,12 +96,17 @@ public RolloverAction(StreamInput in) throws IOException {
...
     public void writeTo(StreamOutput out) throws IOException {
         boolean hasMaxSize = maxSize != null;
-        out.writeBoolean(hasMaxSize);
+        boolean hasMaxPrimaryShardSize = maxPrimaryShardSize != null;
         if (hasMaxSize) {
+            out.writeBoolean(true);
             maxSize.writeTo(out);
+        } else if (hasMaxPrimaryShardSize && out.getVersion().before(Version.V_7_13_0)) {
+            out.writeBoolean(true);
+            maxPrimaryShardSize.writeTo(out);
```
Comment (dakrone) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/RolloverAction.java:105]: Can you add a comment about why we're doing this here?

**Review by dakrone — APPROVED** [review]
LGTM assuming the BWC tests are happy also, I left one really minor comment

**Comment by martijnvg** [comment]
The rest-compat failure is unrelated: https://gradle-enterprise.elastic.co/s/iyw7o64hrh4ce/tests/:rest-api-spec:yamlRestCompatTest/org.elasticsearch.test.rest.ClientYamlTestSuiteIT/test%20%7Byaml=search%2F110_field_collapsing%2Ffield%20collapsing%20and%20inner_hits%7D#1

Trying again.

**Comment by martijnvg** [comment]
@elasticmachine run elasticsearch-ci/rest-compat

**Comment by martijnvg** [comment]
@elasticmachine update branch

**Pull request merged**